client/src/main/scala/org/http4s/client/middleware/Retry.scala

```diff
@@ -0,0 +1,50 @@
+package org.http4s
+package client
+package middleware
+
+import org.http4s.Status.ResponseClass.Successful
+import org.http4s.{Response, Request, EmptyBody}
+import org.http4s.client.Client
+import scala.concurrent.duration._
+import scala.math.{pow, min, random}
+import scalaz.concurrent.Task
+import scala.language.postfixOps
+
+object Retry {
+
+  def apply(backoff: Task[Int => Option[FiniteDuration]])(client: Client) = new Client {
```

Comment:
Why does `backoff` live in the `Task` monad? The only reason I can think of this being useful is if you want to get _different_ functions (f) for each type you fail but that would seem very magical and not often useful. Of course I could be missing something :wink: .